internal/wrappers/feature-flags.go

```diff
@@ -55,26 +58,70 @@ var FeatureFlagsBaseMap = []CommandFlags{
 }
 
 var FeatureFlags = map[string]bool{}
+var FeatureFlagsSpecific = map[string]bool{}
 
 func HandleFeatureFlags(featureFlagsWrapper FeatureFlagsWrapper) error {
 	allFlags, err := featureFlagsWrapper.GetAll()
 	if err != nil {
-		loadFeatureFlagsDefaultValues()
-
+		LoadFeatureFlagsDefaultValues()
 		return nil
 	}
 
 	loadFeatureFlagsMap(*allFlags)
-
 	return nil
 }
 
+func GetSpecificFeatureFlag(featureFlagsWrapper FeatureFlagsWrapper, flagName string) (*FeatureFlagResponseModel, error) {
+	if value, exists := FeatureFlagsSpecific[flagName]; exists {
+		return &FeatureFlagResponseModel{Name: flagName, Status: value}, nil
+	}
+
+	specificFlag, err := getSpecificFlagWithRetry(featureFlagsWrapper, flagName, maxRetries)
+	if err != nil {
+		UpdateSpecificFeatureFlagMapWithDefault(flagName)
+		return &FeatureFlagResponseModel{Name: flagName, Status: FeatureFlagsSpecific[flagName]}, nil
+	}
+
+	UpdateSpecificFeatureFlagMap(flagName, *specificFlag)
+	return specificFlag, nil
+}
+
+func getSpecificFlagWithRetry(wrapper FeatureFlagsWrapper, flagName string, retries int) (*FeatureFlagResponseModel, error) {
+	var flag *FeatureFlagResponseModel
+	var err error
+
+	for i := 0; i < retries; i++ {
+		flag, err = wrapper.GetSpecificFlag(flagName)
+		if err == nil {
+			return flag, nil
+		}
+	}
+
+	return nil, errors.New("failed to get feature flag after retries")
+}
+
+func UpdateSpecificFeatureFlagMapWithDefault(flagName string) {
+	logger.PrintIfVerbose("Get feature flags failed. Loading defaults...")
+	for _, cmdFlag := range FeatureFlagsBaseMap {
+		for _, flag := range cmdFlag.FeatureFlags {
+			if flag.Name == flagName {
+				FeatureFlagsSpecific[flagName] = flag.Default
+				return
```

Comment:
Should we need this for loops? 
can't it be just FeatureFlagsSpecific[flagName] = FeatureFlags[flagName]

Reply:
Accessing the flag directly by its name would not work here since
FeatureFlagsBaseMap is not a map but a slice.

Therefore, a loop is necessary to search for a specific flag within the nested slices